src/libai/common/Matrix.java

```diff
@@ -224,17 +224,20 @@ public void applyInIdentity(Function f, final Matrix a) {
 	 * Fill the matrix with random values. Alias for fill(true).
 	 */
 	public void fill() {
-		fill(true);
+		fill(true, null);
 	}
 
 	/**
 	 * Fill the matrix with random values. If the fill must be positive call
 	 * with false.
 	 *
 	 * @param signed if allow signed values or not
+	 * @param r The {@link Random} object used to fill the matrix, if {@code null} it will fallback
+	 * to {@link ThreadLocalRandom#current()}
 	 */
-	public void fill(boolean signed) {
-		Random r = new Random(seed);
+	public void fill(boolean signed, Random r) {
```

Comment:
same here, we could create an alias for those cases where null has to be passed explicitly